src/keymap.rs

```diff
@@ -0,0 +1,527 @@
+            KeyPress::Ctrl('E') => Cmd::EndOfLine,
+            KeyPress::Ctrl('F') => {
+                if positive {
+                    Cmd::ForwardChar(n)
+                } else {
+                    Cmd::BackwardChar(n)
+                }
+            }
+            KeyPress::Ctrl('G') |
+            KeyPress::Esc => Cmd::Abort,
+            KeyPress::Ctrl('H') |
+            KeyPress::Backspace => {
+                if positive {
+                    Cmd::BackwardDeleteChar(n)
+                } else {
+                    Cmd::DeleteChar(n)
+                }
+            }
+            KeyPress::Tab => Cmd::Complete,
+            KeyPress::Ctrl('K') => Cmd::KillLine,
+            KeyPress::Ctrl('L') => Cmd::ClearScreen,
+            KeyPress::Ctrl('N') => Cmd::NextHistory,
+            KeyPress::Ctrl('P') => Cmd::PreviousHistory,
+            KeyPress::Meta('\x08') |
+            KeyPress::Meta('\x7f') => {
+                if positive {
+                    Cmd::BackwardKillWord(n, Word::Emacs)
+                } else {
+                    Cmd::KillWord(n, At::End, Word::Emacs)
+                }
+            }
+            KeyPress::Meta('<') => Cmd::BeginningOfHistory,
+            KeyPress::Meta('>') => Cmd::EndOfHistory,
+            KeyPress::Meta('B') => {
+                if positive {
+                    Cmd::BackwardWord(n, Word::Emacs)
+                } else {
+                    Cmd::ForwardWord(n, At::End, Word::Emacs)
+                }
+            }
+            KeyPress::Meta('C') => Cmd::CapitalizeWord,
+            KeyPress::Meta('D') => {
+                if positive {
+                    Cmd::KillWord(n, At::End, Word::Emacs)
+                } else {
+                    Cmd::BackwardKillWord(n, Word::Emacs)
+                }
+            }
+            KeyPress::Meta('F') => {
+                if positive {
+                    Cmd::ForwardWord(n, At::End, Word::Emacs)
+                } else {
+                    Cmd::BackwardWord(n, Word::Emacs)
+                }
+            }
+            KeyPress::Meta('L') => Cmd::DowncaseWord,
+            KeyPress::Meta('T') => Cmd::TransposeWords,
+            KeyPress::Meta('U') => Cmd::UpcaseWord,
+            KeyPress::Meta('Y') => Cmd::YankPop,
+            _ => self.common(key, n, positive),
+        };
+        debug!(target: "rustyline", "Emacs command: {:?}", cmd);
+        Ok(cmd)
+    }
+
+    fn vi_arg_digit<R: RawReader>(&mut self,
+                                  rdr: &mut R,
+                                  config: &Config,
+                                  digit: char)
+                                  -> Result<KeyPress> {
+        self.num_args = digit.to_digit(10).unwrap() as i16;
+        loop {
+            let key = try!(rdr.next_key(config.keyseq_timeout()));
+            match key {
+                KeyPress::Char(digit @ '0'...'9') => {
+                    self.num_args = self.num_args * 10 + digit.to_digit(10).unwrap() as i16;
```

Comment:
This can overflow, might want to use `wrapping_add()`/`wrapping_mul()` or even better `checked_add()`/`checked_mul()`. In debug mode overflow would cause a panic. In release mode, the count would simply be wrong. Though in the event of an overflow, the count is going to be wrong anyway no matter what you do, so overall this is a minor issue.